lighthouse-core/gather/gatherers/unload-listeners.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+
+class UnloadListeners extends Gatherer {
```

Comment:
it does feel weird that this is called the `UnloadListeners` artifact but then also contains pagehide/visibilitychange.

Is there a strong reason you didn't want to a `GlobalListeners`/`WindowListeners` artifact?

It could have a narrow focus to start with these visibility ones but it does go better inline with our "generic artifacts, specific audits" pattern we're pushing

Reply:
> Is there a strong reason you didn't want to a `GlobalListeners`/`WindowListeners` artifact?
> 
> It could have a narrow focus to start with these visibility ones but it does go better inline with our "generic artifacts, specific audits" pattern we're pushing

scared of e.g. some future need for the source of one specific event handler meaning we start saving the source of all the event handlers in the artifact, but we should switch it to one of those, I agree.